NDTensors/src/default_storage.jl

```diff
@@ -0,0 +1,39 @@
+## This is a fil which specifies the default storage type provided some set of parameters
+## The parameters are the element type and storage type
+function default_storage_type(
+  storetype::Type{<:TensorStorage}=Dense,
+  eltype::Type=Float64,
+  datatype::Type{<:AbstractArray}=Vector,
+  index=Tuple{Int},
+)
+  #TODO check if indices are block sparse or not
+  type = nothing
+  try
+    type = storetype{eltype,datatype{eltype}}
+  catch e
+    println("Cannot construct storetype", storetype)
+    throw(e)
+  end
+  return type
+end
+
+default_storage_type(eltype::Type{<:Number}) = default_storage_type(Dense, eltype)
+
+function default_storage_type(datatype::Type{<:AbstractArray})
+  return default_storage_type(Dense, Float64, datatype)
+end
+
+## This function needs to be defined for each vector type
+function default_storage_type(datatype::Type{<:Vector{ElT}}) where {ElT}
+  return default_storage_type(Dense, ElT, Vector)
+end
+
+function default_storage_type(eltype::Type, datatype::Type{<:AbstractArray})
```

Comment:
Is this needed? I would hope that for simplicity we could avoid versions where you input multiple types, at least for now.

Reply:
I have updated this and now have 2 functions
`default_datatype(eltype::Type{<:Number})` and `default_storagetype(datatype::Type{<:AbstractArray{ElT}}) where {ElT}`
In this way, if you call `default_storagetype(Vector{Vector{Float64}})` the code correctly constructs a `Dense{Vector{Float64}, Vector{Vector{Float64}}}` but it is not possible to call `default_datatype(Vector{Float64})` so it might make sense to choose one and use it for both definitions above. I would think it would be more flexible to change the first definition to `default_datatype(eltype::Type)`